### PR TITLE
Add documentation for volatility.

### PR DESCRIPTION
**Pull request by dnwpark**
Related #7692
Related #4617

### CODE REVIEW

**Comment by dnwpark** [comment]
Two lines go over the 80 char limit, but I'm not sure how best to do that with the docs file format.

File: docs/reference/edgeql/volatility.rst [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+successive evaluations. 
+
+Expressions may have one of the following volatilities, in order of increasing
+volatility:
+
+* ``Immutable``: The expression cannot modify the database and is
+    guaranteed to have the same value *in all statements*.
+
+    Pointers maybe be treated as immutable for the purposes of some
+    restrictions. See below for details.
```
Comment (msullivan) [inline docs/reference/edgeql/volatility.rst:17]: I'd drop this from here, I think?

File: docs/reference/edgeql/volatility.rst [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    function plus_insert(x: float64) -> float64
+        using (x + (insert One).val);
+
+Some example operator and function calls:
+
+.. code-block::
+
+    1 + 1:                    Immutable
+    1 + global one:           Stable
+    global one + random():    Unstable
```
Comment (msullivan) [inline docs/reference/edgeql/volatility.rst:101]: Volatile. Unstable isn't a thing :P. (Though it would be a fine name for it.)

**Review by msullivan — APPROVED** [review]
Looks great!

**Pull request merged**